### PR TITLE
Milestone 2: Add Decorators

### PR DESCRIPTION
**Pull request by binyamolango**
# Hi Reviewer 👋

> Here are the changes I implemented to the project

1. Added component interface by creating a class Nameable and implementing a method called correct_name that will raise a NotImplementedError.
2. Turned my Person class to Nameable
 by making sure that my Person class inherits from Nameable
 and has a method correct_name implemented that simply returns the name attribute.
3. Prepared base Decorator
 by inheriting it from Nameable assigning a namable object in the constructor, and implementing the correct_name method that returns the result of the correct_name method of the @nameable.
4. Prepared CapitalizeDecorator and TrimmerDecorator.
5. Added tests to test the decorators.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @nameable.correct_name
+  end
+end
+
+# Concrete Decorator
+class CapitalizeDecorator < PersonDecorator
+  def correct_name
+    super.upcase
+  end
+end
```
Comment (roshan-bajgain) [inline nameable.rb]: - [x] Good job on implementing the CapitalizeDecorator class but when you use the `.upcase` then it makes the whole letter to uppercase so, I would suggest you please use the ``capitalize``: It makes only the first letter capitalize not whole.
```ruby
def correct_name
      @nameable.correct_name.capitalize
    end
```

File: nameable.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+class TrimmerDecorator < PersonDecorator
+  def correct_name
+    max_len = 10
+    if super.length > max_len
+      trimmed_word = super[0..max_len]
+      trimmed_word << '...'
+      trimmed_word
+    else
+      super
+    end
```
Comment (roshan-bajgain) [inline nameable.rb]: - [x] When I run the tests.rb file then I found that it return 11 characters but according to the project requiremenet if  `The name has a maximum of 10 characters. If it's longer it should trim the word.`. Just take the below as example only.
```ruby
def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
```

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @binyamolango, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Almost done ✔️ 

## Required Changes ♻️
- [ ] I would suggest you please use the correct link to you issue page in your README file. It makes you README file more professional.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @binyamolango, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] All requested changes are made ✔️ 
- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**